File: BinaryTree/Leetcode103.java
public class Leetcode103 {
    public List<List<Integer>> zigzagLevelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        if(root == null)return res;
        boolean zig = false;
        Queue<TreeNode> q= new LinkedList<>();
        q.offer(root);
        while(!q.isEmpty()){
            int size = q.size();
            List<Integer> list = new ArrayList<>();
            for(int i = 0;i < size;i++){
                TreeNode node = q.poll();
                if(node.left != null)q.offer(node.left);
                if(node.right != null)q.offer(node.right);
                list.add(node.val);
            }
            if(zig){
                Collections.reverse(list);
            }
            res.add(list);
            zig = !zig;
        }
        return res;
    }
}

File: BinarySearchTree/Python/Leetcode103.py
from queue import Queue
class Leetcode103:
    def zigzagLevelOrder(self, root: TreeNode) -> List[List[int]]:
        res = []
        if root is None:
            return res
        q = Queue()
        q.put(root)
        flag = True
        while not q.empty():
            levelNode = []
            for i in range(q.qsize()):
                n = q.get()
                if n.left is not None: q.put(n.left)
                if n.right is not None: q.put(n.right)
                levelNode.append(n.val)
            if not flag:
                levelNode = list(reversed(levelNode))
            res.append(levelNode)
            flag = not flag
        return res
